.refactored/sciserver-essentials/2.0/Dockerfile

```diff
@@ -0,0 +1,194 @@
+	&& rm -rf /var/cache/yum
+
+USER idies
+
+ARG R_VERSION=4.0.3
+
+RUN curl -L https://cran.rstudio.com/src/base/R-4/R-$R_VERSION.tar.gz -o R.tar.gz \
+	&& tar -xzf R.tar.gz \
+	&& cd R-$R_VERSION \
+	&& ./configure --prefix=/home/idies/R --enable-R-shlib --with-x=no \
+	&& make -j`nproc` \
+	&& make install \
+	&& cd .. \
+	&& rm -rf R.tar.gz R-$R_VERSION
+
+ENV PATH /home/idies/R/bin:$PATH
+
+RUN Rscript -e "install.packages(c('Cairo', 'IRkernel'), repos='https://cloud.r-project.org/')" \
+	&& Rscript -e "IRkernel::installspec(displayname = 'R $R_VERSION')" \
+	&& /bin/bash -c $'printf "options(bitmapType=\'cairo\')\n" >> ~/.Rprofile'
+
+
+# SciScript Python
+
+RUN conda install -n py38 -y \
+		argcomplete \
+		chest \
+		configobj \
+		dill \
+		redis \
+		redis-py \
+		sphinx_rtd_theme \
+		traceback2 \
+		unittest2 \
+	&& conda install -n py38 -y -c conda-forge \
+		grequests \
+	&& conda clean -y --all
+
+RUN git clone https://github.com/sciserver/SciScript-Python.git \
+	&& touch /home/idies/keystone.token \
+	&& cd /home/idies/SciScript-Python \
+	&& git checkout sciserver-v2.0.13 \
+	&& cd py3 \
+	&& (source activate py38 && python setup.py install) \
+	&& rm /home/idies/keystone.token
+
+
+# SciScript R
+
+RUN Rscript -e "install.packages(c('bit64', 'jpeg', 'httr', 'jsonlite', 'data.table'), repos='https://cloud.r-project.org/')"
+
+RUN git clone https://github.com/sciserver/SciScript-R.git \
+	&& touch /home/idies/keystone.token \
+	&& cd /home/idies/SciScript-R \
+	&& git checkout sciserver-v2.0.13 \
+	&& cd .. \
+	&& R CMD build SciScript-R \
+	&& R CMD INSTALL SciServer*.tar.gz \
+	&& rm -rf /home/idies/SciServer*.tar.gz \
+	&& rm /home/idies/keystone.token
+
+
+# Mamba (fast package manager)
+
+RUN conda install -y -c conda-forge mamba \
+        && conda clean -y --all
+
+
+# Tensorflow
+
+RUN mamba install -n py38 -y tensorflow==2.3.0 \
+	&& conda clean -y --all
+
+
+# Link to JupyterLab
+
+COPY --chown=idies lab_extension /home/idies/lab_extension
+
+RUN jupyter nbextension install /home/idies/lab_extension --user \
+	&& jupyter nbextension enable lab_extension/main --user --section='common'
+
+
+# Ensure matplotlib inline rendering by default
+
+RUN ipython profile create \
+	&& printf "\nc.InteractiveShellApp.matplotlib = 'inline'\n" >> /home/idies/.ipython/profile_default/ipython_kernel_config.py
+
+
+# Plotly
+
+RUN mamba install -y -c conda-forge plotly \
+	&& mamba install -n py38 -y -c conda-forge plotly \
+	&& conda clean -y --all
+
+
+# Geo
+
+RUN mamba install -n py38 -y -c conda-forge \
+		geopy \
+		osmnx \
+		geoplot \
+		cartopy \
+		pyepsg \
+		descartes \
+		geos \
+		libspatialindex \
+		libiconv \
+		geopandas \
+		pysal \
+		rtree \
+	&& (source activate py38 && pip install --no-cache-dir \
+		GeobricksProj4ToEPSG \
+		jsonpath-rw-ext \
+		census \
+		us \
+		censusgeocode) \
+	&& conda clean -y --all
+
+USER root
+RUN yum -y install libnsl.so.1 && ln -s /usr/lib64/libnsl.so.2 /usr/lib64/libnsl.so.1
```

Comment:
Any particular reason not to bundle all the yum installs in command?

Reply:
Not really. It's in the Geo section just because this one library and symlinks was a fix for a problem with geopandas.